controllers/auth.js

```diff
@@ -0,0 +1,147 @@
+const jwt = require("jsonwebtoken");
+const User = require("../service/schemas/user");
+
+require("dotenv").config();
+
+const secret = process.env.SECRET;
+
+const register = async (req, res, next) => {
+  const { email, password } = req.body;
+
+  try {
+    const existingUser = await User.findOne({ email });
+    if (existingUser) {
+      return res.status(409).json({ message: "Email in use" });
+    }
+
+    const newUser = new User({ email });
+    newUser.setPassword(password);
+    await newUser.save();
+
+    res.status(201).json({
+      status: "success",
+      user: {
+        email: newUser.email,
+        subscription: newUser.subscription,
+      },
+    });
+  } catch (error) {
+    next(error);
+  }
+};
+
+const login = async (req, res, next) => {
+  try {
+    const { email, password } = req.body;
+    const user = await User.findOne({ email });
+
+    if (!user || !user.validPassword(password)) {
+      return res.status(401).json({ message: "Email or password is wrong" });
+    }
+
+    const payload = {
+      id: user._id,
+      email: user.email,
+    };
+
+    const token = jwt.sign(payload, secret, { expiresIn: "1h" });
```

Comment:
Bez refresh tokenu to osoba bedzie co godzine musiala sie zalogowac. Myślę że warto zwiekszyć nawet na kilka dni :)

Reply:
dalam 3 dni xD